cfgov/v1/models/base.py

```diff
@@ -225,6 +225,43 @@ def get_next_appropriate_siblings(self, hostname, inclusive=False):
     def get_prev_appropriate_siblings(self, hostname, inclusive=False):
         return self.get_appropriate_siblings(hostname=hostname, inclusive=inclusive).filter(path__lte=self.path).order_by('-path')
 
+    def get_context(self, request, *args, **kwargs):
+        context = super(CFGOVPage, self).get_context(request, *args, **kwargs)
+        for hook in hooks.get_hooks('cfgovpage_context_handlers'):
+            hook(self, request, context, *args, **kwargs)
+        return context
+
+    def serve(self, request, *args, **kwargs):
+        """
+        If request is ajax, then return the ajax request handler response, else
+        return the super.
+        """
+        if request.is_ajax():
+            return self.serve_ajax(request)
+
+        return super(CFGOVPage, self).serve(request, *args, **kwargs)
+
+    def serve_ajax(self, request):
+        """
+        Attempts to retreive form_id from the POST request and returns a JSON
+        response.
+
+        If form_id is found, it returns the response from the block method
+        retrieval.
+
+        If form_id is not found it returns an error response.
+        """
```

Comment:
I agree with the status code comment, I'll add that one in. The idea behind the form_id is that a content editor could add multiple forms on a page, so we'd need a way figure out which form should receive the data in the request. 

Since it would be quite difficult to guarantee that a Wagtail page won't have a form in the content, I decided that a form_id attribute in the request was a necessity. I'm open to other suggestions or conclusions though. 

This also doesn't leave out the idea of adding more logic to handle AJAX requests that don't have the form_id attribute, it's just not something we do at the moment.